### PR TITLE
support multi-vpc in synthesis output

### PR DESCRIPTION
**Pull request by olasaadi99**

### CODE REVIEW

**Comment by kyorav** [comment]
When applied to an example that uses a transit gateway the output contains the string `combined-vpc-local-tg1/` instead of the name of the VPC. Example config object attached (taken from the `transit-gateway-multiple` example)
[config_object.json](https://github.com/user-attachments/files/16798787/config_object.json)

**Comment by kyorav** [comment]
I tested on several examples and it looks great

File: pkg/vpcmodel/output.go [hunk 1]
```diff
@@ -301,14 +301,51 @@ func (of *serialOutputFormatter) AggregateVPCsOutput(outputList []*SingleAnalysi
...
+		}
+		if conn.Dst.Type == spec.ResourceTypeExternal {
+			if val, ok := externalsMap[conn.Dst.Name]; ok {
+				conn.Dst.Name = val
+			} else {
+				name := externalString + strconv.Itoa(len(externalsMap))
+				externalsMap[conn.Dst.Name] = name
+				conn.Dst.Name = name
+			}
+		}
```
Comment (zivnevo) [inline pkg/vpcmodel/output.go:343]:
```suggestion
		if conn.Src.Type == spec.ResourceTypeExternal {
			conn.Src.Name = getNewExternalName(conn.Src.Name, externalsMap)
		} else if conn.Dst.Type == spec.ResourceTypeExternal {
			conn.Dst.Name = getNewExternalName(conn.Dst.Name, externalsMap)
		}
```
Avoid code duplication

File: pkg/vpcmodel/output.go [hunk 2]
```diff
@@ -301,14 +301,51 @@ func (of *serialOutputFormatter) AggregateVPCsOutput(outputList []*SingleAnalysi
...
 
+func renameExternals(requiredConnections []spec.SpecRequiredConnectionsElem,
+	externalsMap map[string]string) []spec.SpecRequiredConnectionsElem {
+	connLines := []spec.SpecRequiredConnectionsElem{}
+	for _, conn := range requiredConnections {
+		if conn.Src.Type == spec.ResourceTypeExternal {
+			if val, ok := externalsMap[conn.Src.Name]; ok {
+				conn.Src.Name = val
+			} else {
+				name := externalString + strconv.Itoa(len(externalsMap))
```
Comment (zivnevo) [inline pkg/vpcmodel/output.go:330]: `fmt.Sprintf()` might be better here

**Review by zivnevo — CHANGES_REQUESTED** [review]

**Review by zivnevo — APPROVED** [review]

**Pull request merged**